proposals/extend_pod_security_admission.md

```diff
@@ -0,0 +1,229 @@
+# Extending Pod Security Admission
+
+**Author**: Shuting Zhao (shuting@nirmata.com)
+**Date**: Dec 15th, 2021
+**Update**: June 15th, 2022
+**Abstract**: Using Kyverno to extend Pod Security Admission for Kubernetes.
+
+## Contents
+
+  * [Contents](#contents)
+  * [Introduction](#introduction)
+  * [The Problem](#the-problem)
+  * [Proposed Solution](#proposed-solution)
+    + [Extend PSA via a new rule](#extend-psa-via-a-new-rule)
+    + [Exceptions for container security context](#exceptions-for-container-security-context)
+  * [Alternate Solutions Considered](#alternate-solutions-considered)
+    + [Extend Via Annotations](#extend-via-annotations)
+      - [Step 1. Enable additional policies for a namespace](#step-1-enable-additional-policies-for-a-namespace)
+      - [Step 2. Fine-grained Policy Control](#step-2-fine-grained-policy-control)
+  * [Next Steps](#next-steps)
+
+<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>
+
+
+## Introduction
+
+[Pod Security admission](https://kubernetes.io/docs/concepts/security/pod-security-admission/) (PSA) is a built-in solution that applies different isolation levels of [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/) for Pods. With the release of Kubernetes v1.23, PSA has entered beta and is enabled by default for 1.23+ clusters. 
+
+For more information on using PSA, see: [how PSA is used in a cluster](https://hackmd.io/C0prhSfdTbSv9d1ag7Ft9g).
+
+This document proposes a solution to extend PSA for finer-grained and flexible policy control.
+
+## The Problem
+
+Once PSA is enabled for namespaces, a configured level of Privileged, Baseline, or Restricted applies to all pods and workloads within the namespace. The level is configured as a label on the namespace. There is no option to select specific pods or control, for granular policies. 
```

Comment:
Suggestion: add that we can set a specific version for each enforcement mode: `pod-security.kubernetes.io/<MODE>-version: <VERSION>` (default: latest)

Reply:
The problem we try to state here is to granularly select applied pods. What you suggested is to configure the applied PSS version via PSA, do you think this is needed while stating the problem?